content/articles/integrate-ml-to-flask-api-and-deploy-to-heroku/index.md

```diff
@@ -0,0 +1,336 @@
+api.add_resource(Records, '/')
+app.run(port=5000, debug=True)
+```
+
+We test the local API with Postman to make sure it works before deploying it. 
+
+![GET request to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/get_request_local.PNG)
+
+After sending a GET request, we get a response showing the JSON format we should send in a POST request. So now we send a POST request with JSON data in the same format shown above. 
+
+![POST request to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/post_request_to_local_api.PNG)
+
+After sending the POST request, we get the model prediction as a response. Let us see what happens if the client sends the wrong JSON data to the API.
+
+![POST request with the wrong format to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/post_wrong_request_to_local_server.PNG)
+
+If we send the wrong JSON data to the API, we get an error message with the 500 status code. However, the client will also receive the correct JSON data format. Since our API is working fine, we can proceed to deploy it.
+
+
+### Deploying to Heroku
+
+The next step is to deploy our API so other developers can have access to it. Next, we will deploy the API to Heroku. Heroku is a cloud platform that allows developers to build, run, and deploy their apps. If you do not have a Heroku account, you will have to create one by clicking [here](https://signup.heroku.com/).
+
+We need to install Git and Heroku CLI (Command Line Interface) on our PC. Heroku uses git to manage the deployment of apps. Click [here](https://git-scm.com/downloads) to download and install git. Use the link [here](https://devcenter.heroku.com/articles/heroku-cli#download-and-install) to download and install Heroku CLI. 
+
+With Heroku CLI, we can use our terminal to deploy. The Heroku CLI allows us to access Heroku from our terminal. With it, we can create and manage applications on Heroku with ease. To access Heroku CLI, we log in to Heroku from our terminal. 
+
+```bash
+heroku login
+```
+
+This command takes us to the Heroku login page on our web browser. If you want to remain in the CLI and log in, use the command below instead.
+
+```bash
+heroku login -i
+```
+
+After successfully logging in, we can proceed. We can initialize a git repo in our project directory after installing git.
+
+```bash
+git init
+```
+
+We need to add two files to our project directory: `Procfile` and `requirements.txt`. The `Procfile` tells Heroku what to do with the code you want to push to their server. The `requirements.txt` contains all the dependencies we need to install to run our API on Heroku. We will be running the API using `Gunicorn`. First, we need to install it. 
+
+`Gunicorn` is a web server that is more powerful than the built-in server that flask gives you. The built-in server in the flask can only handle one user at a time. But `Gunicorn` can deal with many users at once.

```

Comment:
Insert a link to an article that highlights and backs the fact that Gunicorn is better than the in-built server.

Reply:
Done